### PR TITLE
Add recipe for rstcheck

### PR DESCRIPTION
**Pull request by kynan**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by kynan** [comment]
@conda-forge/help-python no idea why the build fails on Windows. Something appears to go wrong linking the entry-point executable?

File: recipes/rstcheck/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 92c4f79256a54270e0402ba16a2f92d0b3c15c8f4410cb9c57127067c215741f
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (scopatz) [inline recipes/rstcheck/meta.yaml:15]: Should this have an entrypoint?

**Comment by scopatz** [comment]
@kynan - It is noarch python, so it only needs to run on linux

File: recipes/rstcheck/meta.yaml (hunk 1 above)
Comment (kynan) [reply recipes/rstcheck/meta.yaml:15]: Yes it should! Thanks :)

**Comment by kynan** [comment]
Adding the `entrypoint` fixed this, thanks Anthony! I believe this is now good to go @conda-forge/help-python

File: recipes/rstcheck/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - pip
+  run:
+    - python
+    - docutils >=0.7
+
+test:
+  imports:
+    - rstcheck
+  commands:
+    - python -m rstcheck --help
```
Comment (marcelotrevisani) [inline recipes/rstcheck/meta.yaml:31]: Can we add ``pip check`` here please?
```suggestion
    - python -m rstcheck --help
    - pip check
  requires:
    - pip
```

Comment (kynan) [reply recipes/rstcheck/meta.yaml:31]: Sorry, didn't see this in time. Added now!

**Pull request merged**